File: CPP/Program/SmartDevice.cpp
#pragma once
#include <iostream>
#include <string>
using namespace std;

class SmartDevice {
private:
    bool wifiEnabled;       // Status koneksi WiFi
    string networkName;     // Nama jaringan yang terhubung
    
public:
    // Constructor
    SmartDevice(bool wifiEnabled = false, string networkName = "") {
        this->wifiEnabled = wifiEnabled;
        this->networkName = networkName;
    }
    
    // Setter
    void setWifiEnabled(bool wifiEnabled) { this->wifiEnabled = wifiEnabled; }
    void setNetworkName(string networkName) { this->networkName = networkName; }
    
    // Getter
    bool getWifiEnabled() { return wifiEnabled; }
    string getNetworkName() { return networkName; }
    
    // Tampilkan Info
    void tampilkanInfoSmart() {
        cout << "Wifi: " << (wifiEnabled ? "Aktif" : "Nonaktif") << " | Jaringan: " << (networkName.empty() ? "Tidak terhubung" : networkName);
    }

    // Destructor
    ~SmartDevice() {}
};

File: Java/Program/SmartDevice.java
public class SmartDevice {
    private boolean wifiEnabled;
    private String networkName;

    // Constructor
    public SmartDevice() {
        this.wifiEnabled = false;
        this.networkName = "";
    }

    public SmartDevice(boolean wifiEnabled, String networkName) {
        this.wifiEnabled = wifiEnabled;
        this.networkName = networkName;
    }

    // Setter methods
    public void setWifiEnabled(boolean wifiEnabled) { this.wifiEnabled = wifiEnabled; }
    public void setNetworkName(String networkName) { this.networkName = networkName; }

    // Getter methods
    public boolean getWifiEnabled() { return wifiEnabled; }
    public String getNetworkName() { return networkName; }

    // Tampilkan Info
    public String tampilkanInfoSmart() {
        return "WiFi: " + (wifiEnabled ? "Aktif" : "Nonaktif") + " | Jaringan: " + (networkName.isEmpty() ? "Tidak terhubung" : networkName);
    }
}